yt/utilities/hierarchy_inspection.py

```diff
@@ -39,4 +39,4 @@ def find_lowest_subclasses(candidates: List[Type]) -> List[Type]:
 
     count = reduce(lambda x, y: x + y, counters)
 
-    return [x for x in count.keys() if count[x] == 1]
+    return [x for x in candidates if count[x] == 1]
```

Comment:
while seemingly harmless, I'd rather keep this out of the PR, because it _might_ complicate backporting operations. Unless I am not understanding this correctly and it's not a pure noop ?

Reply:
It is not a noop; I'll pull it into a separate PR with more details today. I mostly wanted to make sure all the tests passed with both changes, since I couldn't get the answer tests to work locally.

Reply:
for posterity this is #4404 
Thanks @yut23 !